### DISCUSSION TITLE
convert all images to a specifc hue

### DISCUSSION THREAD

**Discussion by saaraahfar** [ImageMagick/ImageMagick6#208] [General]
### ImageMagick version

6.9.11-60 Q16 x86_64 2021-01-25

### Operating system, version and so on

linux mint 21.1

### Description

I want to convert a bunch of similar image that has different colors with white random paths but slightly different color for background:
(!https://imgbox.com/Si55Q0jI)
(!https://imgbox.com/TxpvIgwk)

what I want is to convert them all to be a single color consistent, but of course just the background color not the white paths

I tried this matrix which convert it to a grey color but when I want to colorize it, it turns to a different color

```
mogrify -format jpg -path ../z -repage -1-4  -color-matrix  \
' 1 0 0
  1 0 0
  1 0 0 '   +level-colors "#997722", -modulate 100,160,100 *.jpg

```

**Comment by fmw42** [comment]
I do not understand. Do you want all images to have the same color or just make the color uniform in any given image?

If the latter, you can replace any non-white color using -fuzz XX% -fill some_color +opaque white, where some_color is the color you want to use to replace everything that is not white (within the fuzz value). The issue is you need to measure or know what the background color is approximately (within the fuzz range). So you could just measure any non-white color in your image.

Please explain in more detail what you are actually trying to do,

**Reply by saaraahfar** [reply]
yes I want to make them uniform, all of the picture to have same exact beige color rather.

**Reply by fmw42** [reply]
What color do you want?

**Reply by saaraahfar** [reply]
normal beige, I mean I can replace the color the problem is I can not figure out the command

**Reply by saaraahfar** [reply]
it doesn't work as excpect, there are a lot of noises, is there anything that can do it like photoshop when you change the hue. in this case we also need to uniform the darkness/lightness of the color so all color be the same.

**Reply by fmw42** [reply]
Can you provide more details and screen snaps on how you do it in Photoshop? Is your photoshop results satisfactory?

**Comment by fmw42** [comment]
These two approaches are what I have so far.

Unix Syntax:

Input:

![img1](https://user-images.githubusercontent.com/13458218/212983648-eae8c9aa-f527-47e5-b5ca-68783f4e240e.jpg)


```
convert img1.jpg \
\( -clone 0 -fill tan -colorize 100 \) \
\( -clone 0 -colorspace HSV -channel 2 -separate +channel -auto-level -threshold 85%  \) \
\( -clone 0 -colorspace HSV -channel 2 -separate +channel -auto-level -threshold 65% -negate \) \
\( -clone 2,3 -compose plus -composite -negate \) \
-delete 2,3 \
-compose over -composite \
img1_result1.png

```

![img1_result1](https://user-images.githubusercontent.com/13458218/212983712-ab42dc23-5bb7-40ef-8632-c789d4b76bbf.png)




```
convert img1.jpg \
\( -clone 0 -fill tan -colorize 100 \) \
\( -clone 0 -colorspace HSV -channel 2 -separate +channel -contrast-stretch 5,5% \
-white-threshold 65% -fill black -opaque white -fill white +opaque black \) \
-compose over -composite \
img1_result2.png
```

![img1_result2](https://user-images.githubusercontent.com/13458218/212983738-99652c14-dffb-4fee-86d3-24af8f53bfcb.png)

**Change "tan" to whatever color you want**

**Reply by fmw42** [reply]
Your images that you provided were all JPG. They have compression artifacts that are causing issues.  Do you have the same images in PNG or TIFF or some uncompressed format?

**Comment by fmw42** [comment]
If you are on IM 7, you could also do

```
magick img1.jpg \
\( +clone -scale 1x1! -set option:color "%[pixel:u.p]" \) \
-fuzz 8% -fill none -opaque "%[color]" -background "tan" -flatten img1_result3.png
```

![img1_result3](https://user-images.githubusercontent.com/13458218/213257338-446c8504-0f19-4641-93e0-1e1ca8933684.png)

**Adjust the fuzz value as desired**

**Comment by snibgo** [comment]
Another possibility, naming your input as waveyWhite.jpg:
```
magick xc:Black xc:tan xc:White +append +repage -write mpr:MAP +delete waveyWhite.jpg -colorspace Gray  mpr:MAP -clut x.png
```

![x](https://user-images.githubusercontent.com/33812368/213895124-61334c24-d17a-45ef-9b77-b7d5d709c5a5.png)

**Comment by maikevictorpereira-blip** [comment]
What imgbox.com is porn?!
